include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -901,13 +858,8 @@ namespace CUDAWrappers
     for (unsigned int i = 0; i < n_colors; ++i)
       {
         n_cells[i] = graph[i].size();
-        helper.setup_cell_arrays(i);
-        typename std::vector<CellFilter>::iterator cell     = graph[i].begin(),
-                                                   end_cell = graph[i].end();
-        for (unsigned int cell_id = 0; cell != end_cell; ++cell, ++cell_id)
-          helper.get_cell_data(*cell, cell_id, partitioner);
-
-        helper.alloc_and_copy_arrays(i);
+        helper.setup(i);
+        helper.fill_data(i, graph[i], partitioner);
```

Comment:
I think it would make sense to merge these two functions.

Reply:
done